### DISCUSSION TITLE
What is the proper mapping configuration for the image in the docs?

### DISCUSSION THREAD

**Discussion by VictorioBerra** [thepirat000/Audit.NET#478] [Q&A]
This is regarding my setup comprised of .NET 6, and EF Core 6 and SQL Server.

![image](https://user-images.githubusercontent.com/2934507/153327002-dc5ab86c-8d45-4f5c-9786-e589c8f80cf7.png)

From what I understand, if you are using a regular `int Id` EF Core will configure the column as a non-null auto-incrementing PK.

If you use this as-is with something like `.Map<Order, AuditOrder>()`, you will get errors throwing saying `Cannot insert explicit value for identity column in table 'AuditOrder' when IDENTITY_INSERT is set to OFF.`.

I notice the tests show an example where you would instead create an `OrderId` on Audit_Order: https://github.com/thepirat000/Audit.NET/blob/master/test/Audit.EntityFramework.Core.UnitTest/EfCoreTests.cs#L131

Is there another way to do this?

**Comment by thepirat000** [comment]
I guess the problem is that you didn't set `IgnoreMatchedProperties()`, so the property `AuditOrder.Id` is automatically set to the value of the audited `Order.Id`

Can you share your configuration code?

**Reply by VictorioBerra** [reply]
@thepirat000 thanks for the reply. I am still in the planning phase and my code is fluid, I am trying different table setups (generic audit table, or table per entity to audit, pros/cons of each?)

If I call IgnoreMatchedProperties, then in your screenshot, I would have to manually .Map() Audit_Order.Number, and Audit_Order.Status to Oder.Number, and Order.Status. And then do the same for OrderItem.

What I am trying to get away from:

If I go the route of an Audit_* table per entity, and I want to match properties automatically, I will have to have an explicit .Map() on every single entity and I will have to manually add a PK to every single entity (Audit_Order.OrderId).

It leads to a ton of boilerplate code to do that.
